codectrl-gui/src/components/details_view_components/code_highlighting.rs

```diff
@@ -6,10 +6,19 @@ use syntect::{
     util::LinesWithEndings,
 };
 
+fn lang_to_short(lang: &str) -> &str {
+    // Add more language here
+    match lang {
+        "rust" => "rs",
+        "python" => "py",
+        _ => "",
+    }
+}
+
 pub fn code_highlighter(code: &str, lang: &str) -> egui::text::LayoutJob {
     let ps = SyntaxSet::load_defaults_newlines();
     let ts = ThemeSet::load_defaults();
-    let syntax = ps.find_syntax_by_extension("py").unwrap();
+    let syntax = ps.find_syntax_by_extension(lang_to_short(lang)).unwrap();
```

Comment:
https://docs.rs/syntect/4.6.0/syntect/parsing/struct.SyntaxSet.html#method.find_syntax_by_name

You should be able to use that and just pass `lang` into it, rather than creating a new function for it.